scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat

```diff
@@ -382,12 +382,168 @@ object a {
 >>>
 object a {
   val b = c match {
-    case true if d.map({
+    case true if d.map {
+          case true  => true
+          case false => false
+        } =>
+  }
+}
+<<< #2194 5 infix with parens
+object a {
+  val b = c match {
+    case true if (d map(({
+      case true  => true
+      case false => false
+    }))) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case true if d map {
+          case true  => true
+          case false => false
+        } =>
+  }
+}
+<<< #2194 5 infix with parens and extra term
+object a {
+  val b = c match {
+    case true if true && (d map({
+      case true  => true
+      case false => false
+    })) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case true if true && (d map {
+          case true  => true
+          case false => false
+        }) =>
+  }
+}
+<<< #2194 6 match with parens and scala2
+object a {
+  val b = c match {
+    case true if (d match {
+      case true  => true
+      case false => false
+    }) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case true if (d match {
           case true  => true
           case false => false
         }) =>
   }
 }
+<<< #2194 6 match with parens and scala3
+runner.dialect = scala3
+===
+object a {
+  val b = c match {
+    case true if (d match (({
+      case true  => true
+      case false => false
+    }))) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case true if d match {
+          case true  => true
+          case false => false
+        } =>
+  }
+}
+<<< #2194 6 match with parens and no braces
+runner.dialect = scala3
+===
+object a {
+  val b = c match {
+    case true if (d match
+      case true  => true
+      case false => false
+    ) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case true if d match
+          case true  => true
+          case false => false
+        =>
+  }
+}
+<<< #2194 6 match with parens and dot
+runner.dialect = scala3
+===
+object a {
+  val b = c + (d.match{
+      case true  => true
+      case false => false
+  })
+  val b = c + (d.match{
+      case true  => true
+      case false => false
+  }) + e
+  val b = c + (d match{
+      case true  => true
+      case false => false
+  })
+}
+>>>
+object a {
+  val b = c + d.match {
+    case true  => true
+    case false => false
+  }
+  val b = c + d.match {
+    case true  => true
+    case false => false
+  } + e
+  val b = c + (d match {
+    case true  => true
+    case false => false
+  })
+}
+<<< #2194 6 match with parens, dot and no braces
+runner.dialect = scala3
+===
+object a {
+  val b = c + (d.match
+      case true  => true
+      case false => false)
+  val b = c + (d.match
+      case true  => true
+      case false => false) + e
+  val b = c + (d match
+      case true  => true
+      case false => false
+  )
+}
+>>>
+object a {
+  val b = c + (d.match
+    case true  => true
+    case false => false
+  )
+  val b = c + (d.match
```

Comment:
And also here.

Reply:
if you remove parens, what would the result be? it tried to do
```
val b = c + d.match
   case true => true
   case false => false
   + e
```
is that valid? if it's not, parens are needed. if it is, at what level should the indentation be for `+ e`?

Reply:
I think it would need to be:
```
val b = c + d.match
   case true => true
   case false => false
+ e
```

Reply:
That's working for me in Dotty

Reply:
Wait no, scratch that. It will wound up in the case. We need parens here.